roomserver/storage/sqlite3/storage.go

```diff
@@ -90,8 +93,18 @@ func (d *Database) StoreEvent(
 			}
 		}
 
-		// TODO: Room version here
-		if roomNID, err = d.assignRoomNID(ctx, txn, event.RoomID(), "1"); err != nil {
+		// Get the default room version. If the client doesn't supply a room_version
+		// then we will use our configured default to create the room.
+		// https://matrix.org/docs/spec/client_server/r0.6.0#post-matrix-client-r0-createroom
+		// Note that the below logic depends on the m.room.create event being the
+		// first event that is persisted to the database when creating or joining a
+		// room.
+		var roomVersion gomatrixserverlib.RoomVersion
+		if roomVersion, err = extractRoomVersionFromCreateEvent(event); err != nil {
+			return err
+		}
+
+		if roomNID, err = d.assignRoomNID(ctx, txn, event.RoomID(), roomVersion); err != nil {
```

Comment:
StoreEvent now has a reasonably significant change when handling create events (the first event) vs anything else. I would be tempted to factor this logic out, and keep StoreEvent for the common case where the room NID + version is set.

Reply:
e.g `StoreInitialEvent(Event, RoomVersion)`

Reply:
So this (and the previous point) are interesting because `StoreEvent` works on the assumption that `assignRoomNID` is idempotent - if a NID already exists, it just returns it. We can remodel into two separate functions.

Reply:
Fewer db calls ftw? :)

Reply:
This doesn't have to happen in this PR.